fluidefi/fluidefi-caspernet-analytics-tools
# Feature/lp hourly summarizer

### Summary of changes:
- Added hourlySummarizer class with all summarizations as methods
- Added unit tests for the summarization methods.
- Added sql docs.

### Steps for testing: 
- You should have already some test data from the previous services (aggregator) 
- Create `hourly_data` table by running this SQL DDL query 
```
CREATE TABLE hourly_data (
    id SERIAL PRIMARY KEY,
    address VARCHAR(100) NOT NULL,
    open_timestamp_utc TIMESTAMP WITH TIME ZONE NOT NULL,
    close_timestamp_utc TIMESTAMP WITH TIME ZONE NOT NULL,
    close_reserves_0 DOUBLE PRECISION NOT NULL,
    close_reserves_1 DOUBLE PRECISION NOT NULL,
    num_swaps_0 BIGINT NOT NULL,
    num_swaps_1 BIGINT NOT NULL,
    num_mints BIGINT NOT NULL,
    num_burns BIGINT NOT NULL,
    mints_0 DOUBLE PRECISION NOT NULL,
    mints_1 DOUBLE PRECISION NOT NULL,
    burns_0 DOUBLE PRECISION NOT NULL,
    burns_1 DOUBLE PRECISION NOT NULL,
    volume_0 DOUBLE PRECISION NOT NULL,
    volume_1 DOUBLE PRECISION NOT NULL,
    max_block INTEGER NOT NULL,
    close_lp_token_supply NUMERIC(155, 0) NOT NULL
);
```
- Run the main.py:
`python main.py`

### How to run unit tests:
`python -m unittest tests/test_hourly_summarizer.py`
